llvm/lib/Support/Compression.cpp

```diff
@@ -202,6 +207,71 @@ Error zstd::decompress(ArrayRef<uint8_t> Input,
   return E;
 }
 
+static void zstdContextFree(ZSTD_DCtx *Ctx) {
+  size_t RC = ::ZSTD_freeDCtx(Ctx);
+  assert(::ZSTD_isError(RC) == 0);
+}
+
+Error zstd::decompress(ArrayRef<uint8_t> Input,
+                       SmallVectorImpl<uint8_t> &Output) {
+  size_t Res = 0;
+  std::unique_ptr<::ZSTD_DCtx, void (*)(::ZSTD_DCtx *)> Ctx(::ZSTD_createDCtx(),
+                                                            zstdContextFree);
+  Res = ::ZSTD_initDStream(&*Ctx);
+  if (::ZSTD_isError(Res))
+    return make_error<StringError>(::ZSTD_getErrorName(Res),
+                                   inconvertibleErrorCode());
+
+  // Create a buffer, this will be filled up repeatedly by
+  // ZSTD_decompressStream and appended to Output
+  SmallVector<char> Buffer;
```

Comment:
The naming is very confusing. `Output`, `Buffer`, `OutputBuffer`.
`Output` => `Decompressed`.
`Buffer` => `BufferVector`
`OutputBuffer` => `ZSTDBuffer`

Reply:
Eh, I know, my intention was for the argument to be consistently named with the rest of the `zstd::decompress` functions

Reply:
> Eh, I know, my intention was for the argument to be consistently named with the rest of the zstd::decompress functions.

At the following at the top of the function to alias the variable:

```c++
auto &Decompressed = Output;
```